pretty_gpx/drawing/drawing_figure.py

```diff
@@ -101,3 +101,90 @@ def draw(self,
 
```

Comment:
Thanks, it would be nice to have this city background for urban runs ! :)

It would be probably more effective if we split the work.
I'd suggest for you to focus on generating a pretty city image background.
Then, I can handle the cache mechanism and the background mode switch between both drawing figures to insert your background generator (slightly different indeed because it looks like it would remain vectorized and not rasterized like the mountain background)

Reply:
try to pay attention to how to leverage the background color for the city background, to keep it aesthetic

![image](https://github.com/user-attachments/assets/063a0e66-ba3a-46aa-afdb-ef3aa8306471)

Reply:
![parisv2](https://github.com/user-attachments/assets/55784a5d-c6f9-47de-81da-46c690de55a8)
I was thinking of this combination of color (first)
Background white, gpx red and roads black

I find this pretty

I am gonna try different possibilities

Reply:
![MPT](https://github.com/user-attachments/assets/27fb4aeb-d772-4378-86aa-d444bec27322)
Upgrading the bounding algo + adding text
Probably need to save the roads image in a cache because it takes a while to regenrate the graph is only editing text as it also plot again the roads (up to 1m in some cases)

Reply:
I don't really like how the text area looks like

However the urban background with all the roads is great!!
Yes definitely, there are too many roads to re-draw everytime, cache will be required. LineCollections are also more efficient with matplotlib

Reply:
I've set up a linter and type-checker, and did a lot of refactoring

I'll try to have a look at `drawing/drawing_city.py` this week end :)

Reply:
> I don't really like how the text area looks like
> 
> However the urban background with all the roads is great!!
> 
> Yes definitely, there are too many roads to re-draw everytime, cache will be required. LineCollections are also more efficient with matplotlib

Thank you for the idea I did not know LineCollection, I implement it in a small fix this morning, it divides the time needed by 2